### PR TITLE
Separate variance and universe fields in inductives.

### PR DESCRIPTION
**Pull request by SkySkimmer**
I think the usage looks cleaner this way.

Overlays: https://github.com/LPCIC/coq-elpi/pull/46 https://github.com/mattam82/Coq-Equations/pull/174 https://github.com/Mtac2/Mtac2/pull/157 https://github.com/coq-community/paramcoq/pull/21

### CODE REVIEW

**Comment by ppedrot** [comment]
It does look slightly cleaner, but it's quite a big patch though.

**Comment by SkySkimmer** [comment]
>It does look slightly cleaner, but it's quite a big patch though.

Net -280 lines is quite small from a certain point of view ;)

**Comment by ppedrot** [comment]
Fair enough, I am convinced.

File: kernel/declarations.ml [hunk 1]
```diff
@@ -52,9 +52,9 @@ type constant_def =
   | Def of constr Mod_subst.substituted   (** or a transparent global definition *)
   | OpaqueDef of Opaqueproof.opaque       (** or an opaque global definition *)
 
-type constant_universes =
-  | Monomorphic_const of Univ.ContextSet.t
-  | Polymorphic_const of Univ.AUContext.t
+type universes =
+  | Monomorphic of Univ.ContextSet.t
+  | Polymorphic of Univ.AUContext.t
```
Comment (ppedrot) [inline kernel/declarations.ml]: The name of these constructors is not very greppable. I would rather call them e.g. `Monomorphic_decl` or something.

**Comment by SkySkimmer** [comment]
Because of the confusion with the .v file Monomorphic?

Gaëtan Gilbert

**Comment by ppedrot** [comment]
Because of the confusion with the corresponding flags in the vernacular language, specifically, and more generally because these words are also used in the English comments in the source code here and there.

**Comment by SkySkimmer** [comment]
@ppedrot are you taking this one?

**Comment by ppedrot** [comment]
@SkySkimmer can you rebase again? I'll merge after a clean CI run.

**Pull request merged**